drivers/android-driver/src/main/java/app/cash/sqldelight/driver/android/AndroidSqliteDriver.kt

```diff
@@ -301,12 +312,23 @@ private class AndroidQuery(
 
   override fun toString() = sql
 
-  override fun close() { }
+  override fun close() {}
 }
 
 private class AndroidCursor(
   private val cursor: Cursor,
+  windowSizeBytes: Long?,
 ) : SqlCursor {
+  init {
+    if (
+      Build.VERSION.SDK_INT >= Build.VERSION_CODES.P &&
+      windowSizeBytes != null &&
+      cursor is AbstractWindowedCursor
+    ) {
+      cursor.window = CursorWindow(null, windowSizeBytes)
```

Comment:
Can you move this line into another class so that its code verification is deferred unless needed.

You can read about why here: https://github.com/androidx/androidx/blob/androidx-main/docs/api_guidelines/compat.md#checking-device-sdk-version-compat-sdk